src/process/script.h

```diff
@@ -0,0 +1,257 @@
+#include "buffer.h"
+#include "../types.h"
+
+// 256 is BINPRM_BUF_SIZE starting on kernels 5.1+.
+#define BINPRM_BUF_SIZE 256
+
+// the kernel deliberately fails if it has to rewrite it more than 4 times
+#define MAX_INTERPRETERS 4
+
+#define PATH_MAX 4096
+
+typedef struct
+{
+  struct {
+    // relative path to the script. This string has already been
+    // copied into kernel space and it is not freed during
+    // interpreter finding thus making it safe to use directly
+    char *path;
+    file_info_t identity;
+  } file;
+
+  file_info_t interpreters[MAX_INTERPRETERS];
+} script_t;
+
+// A map containing script information for currently running exec*
+struct bpf_map_def SEC("maps/scripts") scripts = {
+  .type = BPF_MAP_TYPE_LRU_HASH,
+  .key_size = sizeof(u32),
+  .value_size = sizeof(script_t),
+  .max_entries = 1024, // TODO: should we expand oxidebpf to allow scaling based on # cpus?
```

Comment:
IMO yes, being able to be nicer to smaller low-core-count systems while still being able to handle high-core-count ones seems worth a bit of effort to me. But probably for after the basic implementation is working.